File: solutions/cpp/pangram/1/pangram.cpp
#include "pangram.h"

#include <cctype>
#include <cstdint>

bool pangram::is_pangram(const std::string& str)
{
  std::uint_fast32_t histogram = 0;
  for(const auto c : str) if(std::isalpha(c))
  {
    histogram |= 1 << (std::tolower(c) - 'a');
  }
  return histogram == (1 << 26) - 1;
}

File: solutions/go/pangram/1/pangram.go
package pangram

import "strings"

func IsPangram(input string) bool {
	for _, character := range "abcdefghijklmnopqrstuvwxyz" {
		if !strings.Contains(strings.ToLower(input), string(character)) {
			return false
		}
	}
	return true
}
